### PR TITLE
modified checkSensi to use provider cnecs, added loadflowProvider

### PR DESCRIPTION
**Pull request by phiedw**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*

**What is the current behavior?** *(You can also link to an open issue here)*

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change?** *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: sensitivity-computation/src/main/java/com/farao_community/farao/sensitivity_computation/LoadflowProvider.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+/**
+ * To run a systematic sensitivity analysis and evaluate the flows in all states at once,
+ * hades requires sensitivities. We therefore extend RangeActionSensitivityProvider to use
+ * some of its conversion methods, and use a random PST from the network to create "dummy"
+ * sensitivities for each studied cnec.
+ *
+ * @author Philippe Edwards {@literal <philippe.edwards at rte-france.com>}
+ */
+public class LoadflowProvider extends RangeActionSensitivityProvider {
```
Comment (baptsegu) [inline sensitivity-computation/src/main/java/com/farao_community/farao/sensitivity_computation/LoadflowProvider.java:26]: shouldn't it be the opposite ? the RangeActionSensitivityProvider which extends the LoadFlowProvider.

As the List of RangeAction of the RangeActionSensitivityProvider is not used in this class.

Comment (baptsegu) [reply sensitivity-computation/src/main/java/com/farao_community/farao/sensitivity_computation/LoadflowProvider.java:26]: and functionnaly it seems strange to have access here to some methods of the RangeActionSensitivityProvider, such as : addSensitivityFactors(Set<RangeAction> rangeActions, Set<Cnec> cnecs)

File: sensitivity-computation/src/main/java/com/farao_community/farao/sensitivity_computation/RangeActionSensitivityProvider.java [hunk 2]
```diff
@@ -68,7 +68,7 @@ private boolean willBeKeptInSensi(Generator gen) {
         return gen.getTerminal().isConnected() && gen.getTerminal().getBusBreakerView().getBus().isInMainSynchronousComponent();
     }
 
-    private SensitivityVariable defaultSensitivityVariable(Network network) {
+    protected SensitivityVariable defaultSensitivityVariable(Network network) {
```
Comment (baptsegu) [inline sensitivity-computation/src/main/java/com/farao_community/farao/sensitivity_computation/RangeActionSensitivityProvider.java:71]: could be then moved to the LoadFlowProvider is this class now extends LoadFlowProvider

**Review by baptsegu — CHANGES_REQUESTED** [review]

File: sensitivity-computation/src/main/java/com/farao_community/farao/sensitivity_computation/LoadflowProvider.java (hunk 1 above)
Comment (phiedw) [reply sensitivity-computation/src/main/java/com/farao_community/farao/sensitivity_computation/LoadflowProvider.java]: You're right, i switched the classes around

File: sensitivity-computation/src/main/java/com/farao_community/farao/sensitivity_computation/RangeActionSensitivityProvider.java (hunk 2 above)
Comment (phiedw) [reply sensitivity-computation/src/main/java/com/farao_community/farao/sensitivity_computation/RangeActionSensitivityProvider.java]: yeah I had to move a few methods

**Review by baptsegu — APPROVED** [review]

**Pull request merged**